python/tk_multi_publish2/api/plugins/publish_plugin_instance.py

```diff
@@ -208,22 +208,28 @@ def run_finalize(self, settings, item):
     ############################################################################
     # ui methods
 
-    def run_create_settings_widget(self, parent):
+    def run_create_settings_widget(self, parent, items):
         """
         Creates a custom widget to edit a plugin's settings.
 
         .. note:: This method is a no-op if running without a UI present
 
         :param parent: Parent widget
         :type parent: :class:`QtGui.QWidget`
+        :param items: A list of PublishItems the selected tasks are parented to.
         """
 
         # nothing to do if running without a UI
         if not sgtk.platform.current_engine().has_ui:
             return None
 
         with self._handle_plugin_error(None, "Error laying out widgets: %s"):
-            return self._hook_instance.create_settings_widget(parent)
+            try:
+                return self._hook_instance.create_settings_widget(parent, items)
```

Comment:
This code would catch any TypeError, even those that were raised for other reasons. You may want to look into [getargspec](https://docs.python.org/2.7/library/inspect.html#inspect.getargspec)

Reply:
Yeah I was going to use that approach, but it is deprecated in Python 3

Reply:
I can look into it more

Reply:
OK I've updated it to use inspect, and handle Python 2 vs 3 differences.